### PR TITLE
Upgrade Vite to beta.10

### LINKED ISSUE #67: Upgrade Vite to beta.10

### PR DESCRIPTION
**Pull request by jamesopstad**
resolves #67

This makes the necessary changes to upgrade us to `6.0.0-beta.10`. I've also added tests to ensure the `HotChannel` is working.

I few things surprised me along the way. One is that errors that occur in `HotChannel` no longer seem to propagate. This means that while the issue in #64 is still present, it is no longer visible to the user (sending events just fails silently). I'll raise this with the Vite team.

### CODE REVIEW

File: playground/hot-channel/__tests__/hot-channel.spec.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { expect, test } from 'vitest';
+import { isBuild, serverLogs } from '../../__test-utils__';
+
+test.runIf(!isBuild)('client receives custom events', async () => {
+	expect(serverLogs.info.join()).toContain('__server-event-data-received__');
+});
+
+test.runIf(!isBuild)('server receives custom events', async () => {
+	console.log('serverLogs', serverLogs);
```
Comment (dario-piotrowicz) [inline playground/hot-channel/__tests__/hot-channel.spec.ts:9]:
```suggestion
```

File: packages/vite-plugin-cloudflare/src/runner/module-runner.ts [hunk 2]
```diff
@@ -13,16 +15,25 @@ export async function createModuleRunner(
...
+	});
+
 	moduleRunner = new ModuleRunner(
 		{
 			root: env.__VITE_ROOT__,
 			sourcemapInterceptor: 'prepareStackTrace',
 			transport: {
-				async fetchModule(...args) {
-					const response = await env.__VITE_FETCH_MODULE__.fetch(
+				...transport,
```
Comment (dario-piotrowicz) [inline packages/vite-plugin-cloudflare/src/runner/module-runner.ts:31]: I think this is the bit I missed in my PR 👀

**Review by dario-piotrowicz — APPROVED** [review]
Looks great to me! 😄

File: playground/hot-channel/__tests__/hot-channel.spec.ts (hunk 1 above)
Comment (jamesopstad) [reply playground/hot-channel/__tests__/hot-channel.spec.ts]: 4322cb3

**Pull request merged**